nav2_gps_waypoint_follower_demo/src/gps_waypoint_collector.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2020 Fetullah Atas, Norwegian University of Life Sciences
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+#include "nav2_gps_waypoint_follower_demo/gps_waypoint_collector.hpp"
+
+namespace nav2_gps_waypoint_follower_demo
+{
+
+GPSWaypointCollector::GPSWaypointCollector()
```

Comment:
GPS Waypoint Publisher probably seems like a better name. You could also publish the oriented GPS message too rather than only logging it. That might be helpful so that they can be stored to a vector and then dumped to a file for use.

Reply:
Publishing will make it hard for user as he wont have control on when to trigger publisher. The logging is simple, the user can go to desired location then trigger just by hitting a Enter and receive the current waypoint at terminal.  But storing at vector after each keyboard hit makes sense

Reply:
Well the name "Collector" isn't really accurate since its not collecting things, its just logging them. So maby `GPSWaypointLogger` is a better name?

Reply:
I meant publish on a topic as well as log, so that both are around.